packages/vats/test/bootstrapTests/test-vaults-integration.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * @file Bootstrap test integration vaults with smart-wallet
+ */
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import { Fail } from '@agoric/assert';
+import { E } from '@endo/captp';
+import { makeAgoricNamesRemotesFromFakeStorage } from '../../tools/board-utils.js';
+import { makeSwingsetTestKit, makeWalletFactoryDriver } from './supports.js';
+
+/**
+ * @type {import('ava').TestFn<Awaited<ReturnType<typeof makeDefaultTestContext>>>}
+ */
+const test = anyTest;
+
+const makeDefaultTestContext = async t => {
+  console.time('DefaultTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t);
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { messageVat } = runUtils;
+
+  // Wait for IbcATOM to make it into agoricNames
+  await messageVat('bootstrap', 'consumeItem', ['vaultFactoryKit']);
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  const walletDriver = await makeWalletFactoryDriver(runUtils, storage);
+  console.timeLog('DefaultTestContext', 'walletDriver');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(
+    swingsetTestKit.storage,
+  );
+  agoricNamesRemotes.brand.IbcATOM || Fail`IbcATOM missing from agoricNames`;
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  console.timeEnd('DefaultTestContext');
+
+  return { ...swingsetTestKit, agoricNamesRemotes, walletDriver };
+};
+
+test.before(async t => {
+  t.context = await makeDefaultTestContext(t);
+});
+test.after(async t => {
+  // not strictly necessary but conveys that we keep the controller around for the whole test file
+  await E(t.context.controller).shutdown();
+});
+
+test('metrics path', async t => {
+  const { messageVat, messageVatObject, awaitVatObject } = t.context.runUtils;
+  // example of awaitVatObject
+  const vaultFactoryKit = await messageVat('bootstrap', 'consumeItem', [
+    'vaultFactoryKit',
+  ]);
+  const vfTopics = await messageVatObject(
+    vaultFactoryKit.publicFacet,
+    'getPublicTopics',
+  );
+  const vfMetricsPath = await awaitVatObject(vfTopics.metrics, ['storagePath']);
+  t.is(vfMetricsPath, 'published.vaultFactory.metrics');
+});
+
+test('open vault', async t => {
+  console.time('open vault');
+  const { runUtils, storage } = t.context;
+
+  const factoryDriver = await makeWalletFactoryDriver(runUtils, storage);
+  console.timeLog('open vault', 'made walletDriver');
+
+  const wd = await factoryDriver.provideSmartWallet('agoric1open');
+
+  const { agoricNamesRemotes } = t.context;
+  await wd.executeOffer({
```

Comment:
static typechecking this big offer expression would be nifty

Reply:
it checks the types of the properties, but for some reason it doesn't check the top level properties. gonna punt on that.